openmc/deplete/abc.py

```diff
@@ -364,6 +365,219 @@ def nuclides(self, nuclides):
         self._nuclides = nuclides
 
 
+class FissionYieldHelper(ABC):
+    """Abstract class for processing energy dependent fission yields
+
+    Parameters
+    ----------
+    chain_nuclides : iterable of openmc.deplete.Nuclide
+        Nuclides tracked in the depletion chain. Not necessary
+        that all have yield data.
+
+    Attributes
+    ----------
+    n_bmats : int
+        Number of burnable materials tracked in the problem
+    constant_yields : dict of str to :class:`openmc.deplete.FissionYield`
+        Fission yields for all nuclides that only have one set of
+        fission yield data. Can be accessed as ``{parent: {product: yield}}``
+    """
+
+    def __init__(self, chain_nuclides):
+        self._chain_nuclides = {}
+        self._constant_yields = {}
+
+        # Get all nuclides with fission yield data
+        for nuc in chain_nuclides:
+            if len(nuc.yield_data) == 1:
+                self._constant_yields[nuc.name] = (
+                    nuc.yield_data[nuc.yield_energies[0]])
+            elif len(nuc.yield_data) > 1:
+                self._chain_nuclides[nuc.name] = nuc
+        self._chain_set = set(self._chain_nuclides) | set(self._constant_yields)
+
+    @property
+    def constant_yields(self):
+        out = {}
+        for key, sub in self._constant_yields.items():
+            out[key] = sub.copy()
+        return out
```

Comment:
A copy is used because someone could modify the constant yields and potentially mess up the weighted yields. For example, it is possible to insert a nuclide into `_constant_yields` if we don't copy, and that could overwrite what happens in `weighted_yields`. This is unlikely, as these helpers are meant to be "private" attributes on the Operator.

`copy.deepcopy`  could be used to capture both layers of the nested dictionary